bin/ODiffBin.re

```diff
@@ -10,15 +26,55 @@ let diffPath =
 let base =
   Arg.(
     value
-    & pos(0, file, "")
-    & info([], ~docv="BASE", ~doc="Path to base image")
+    & pos(0, underscore_or(non_dir_file), "_")
+    & info(
+        [],
+        ~docv="BASE",
+        ~doc=
+          "Path to base image (or \"_\" when you want to provide the base image from stdin)",
+      )
+  );
+
+let baseType =
+  Arg.(
+    value
+    & opt(enum([("auto", `auto), ...supported_formats]), `auto)
+    & info(
+        ["base-type"],
+        ~docv="FORMAT",
+        ~doc=
+          Printf.sprintf(
+            "The type of the base image (required to be not auto when a buffer is used as input).\nSupported values are: auto,%s",
+            supported_formats |> List.map(fst) |> String.concat(","),
+          ),
+      )
```

Comment:
You are now able to explicitly set the type of the provided input. This is required for buffer inputs.
We could also try to read the type from the images header, but I think this would be too much work.